background-processor/src/lib.rs

```diff
@@ -68,28 +100,28 @@ impl BackgroundProcessor {
 	/// [`ChannelManager::write`]: lightning::ln::channelmanager::ChannelManager#impl-Writeable
 	/// [`FilesystemPersister::persist_manager`]: lightning_persister::FilesystemPersister::persist_manager
 	pub fn start<
-		PM, Signer,
+		Signer: 'static + Sign,
 		M: 'static + Deref + Send + Sync,
 		T: 'static + Deref + Send + Sync,
 		K: 'static + Deref + Send + Sync,
 		F: 'static + Deref + Send + Sync,
 		L: 'static + Deref + Send + Sync,
 		Descriptor: 'static + SocketDescriptor + Send + Sync,
-		CM: 'static + Deref + Send + Sync,
-		RM: 'static + Deref + Send + Sync
-	>(
-		persist_channel_manager: PM, channel_manager: Arc<ChannelManager<Signer, M, T, K, F, L>>,
-		peer_manager: Arc<PeerManager<Descriptor, CM, RM, L>>, logger: L,
-	) -> Self where
-		Signer: 'static + Sign,
+		CMH: 'static + Deref + Send + Sync,
+		RMH: 'static + Deref + Send + Sync,
+		CMP: 'static + Send + ChannelManagerPersister<Signer, M, T, K, F, L>,
+		CM: 'static + Deref<Target = ChannelManager<Signer, M, T, K, F, L>> + Send + Sync,
+		PM: 'static + Deref<Target = PeerManager<Descriptor, CMH, RMH, L>> + Send + Sync,
+	>
+	(handler: CMP, channel_manager: CM, peer_manager: PM, logger: L) -> Self
```

Comment:
So I did handler because (hopefully very soon) we'll put `handle_events(events: Vec<Event>)` in the same trait and then it will "handle" the background events. Can rename it for now, but hopefully we can migrate soon.

Reply:
Would it be worth re-naming the `ChannelManagerPersister` trait then to be a little more generic? Happy to leave to a follow-up to bikeshed on. :)

Reply:
Yea, I was just figuring we'd leave it as a followup.